snapcraft/cli/lifecycle.py

```diff
@@ -65,6 +66,8 @@ def _execute(
                     instance.pack_project(output=output)
                 else:
                     instance.execute_step(step)
+                if shell_after:
+                    instance.shell()
```

Comment:
This is a little surprising. What happens if the shell exits with error status, at  this point?
I'd expect the shell_after check to be done after we'd cleared the `try`/`except` block -- if necessary with its own exception handler.

Reply:
it will raise a proper exception that is captured correctly in our exit handler

Reply:
is that what you want? the step itself didn't fail, just the shell after. Are the same messages still applicable? Do you want shell and debug to interact in this way?

Reply:
I see your point now. Will fix too